blkin-lib/zipkin_c.c

```diff
@@ -92,6 +92,20 @@ int blkin_init_new_trace(struct blkin_trace *new_trace, const char *service,
 	return res;
 }
 
+int blkin_init_root_info(struct blkin_trace_info *trace_info)
+{
+	int res;
+	if (!trace_info) {
+		res = -EINVAL;
+		goto OUT;
+	}
+	trace_info->span_id = trace_info->trace_id = random_big();
+	trace_info->parent_span_id = 0;
+
+OUT:
+	return res;
+}
+
```

Comment:
Ok. Any reason for not having the same span_id and trace_id in a root span? 

From the [documentation](http://zipkin.io/pages/instrumenting.html) and the [data model example](http://zipkin.io/pages/data_model.html)
> Span Id
> The ID for a particular span. This may or may not be the same as the trace id.

Having the same ID for root spans can be faster. If you agree I can rename it to  blkin_init_new_trace() and take a parameter to set the trace and span ids to the same or a different value, and if not I just use different ids.